backend/kernelCI_app/utils.py

```diff
@@ -318,3 +320,168 @@ def get_grouped_filters(self):
                 grouped_filters[field]['value'].append(value)
 
         return grouped_filters
+
+    def is_build_filtered_out(
+            self, *,
+            duration: Optional[int],
+            valid: Optional[bool],
+            issue_id: Optional[str]
+    ) -> bool:
+        return (
+            len(self.filterBuildValid) > 0
+            and (str(valid).lower() not in self.filterBuildValid)
+        ) or (
+            (self.filterBuildDurationMax is not None or self.filterBuildDurationMin is not None)
+            and duration is None
+        ) or (
+            self.filterBuildDurationMax is not None and (
+                toIntOrDefault(duration, 0) > self.filterBuildDurationMax
+            )
+        ) or (
+            self.filterBuildDurationMin is not None and (
+                toIntOrDefault(duration, 0) < self.filterBuildDurationMin
+            )
+        ) or (
+            len(self.filterIssues["build"]) > 0
+            and (
+                issue_id not in self.filterIssues["build"]
+                or valid is True
+            )
+        )
+
+    def is_record_filtered_out(
+            self, *,
+            hardwares: Optional[List[str]] = None,
+            architecture: Optional[str],
+            compiler: Optional[str],
+            config_name: Optional[str],
+            hasHardware=True
```

Comment:
the `hasHardware` parameter isn't being used when called, I don't think these parameters should assume a default value